statics/js/app.js

```diff
@@ -181,10 +179,18 @@ var app = new Vue({
 
     this.setThemeFromConfig();
 
-    websocket.addConnectHandler(self.onConnected.bind(self));
-    websocket.addDisconnectHandler(self.onDisconnected.bind(self));
-    websocket.addErrorHandler(self.onError.bind(self));
-
+    globalEventHandler.onWebsocketEvent('websocket.connected', self.onConnected.bind(self));
+    globalEventHandler.onWebsocketEvent('websocket.disconnected', self.onDisconnected.bind(self));
+    globalEventHandler.onWebsocketEvent('websocket.error', self.onError.bind(self));
+    this.$store.subscribe(function(mutation) {
+      if (mutation.type === "login") {
+        globalEventHandler.websocket().toggleReconnectMode(true);
+	globalEventHandler.websocket().connect();
+      } else if (mutation.type === 'logout'){
+        globalEventHandler.websocket().toggleReconnectMode(false);
+	globalEventHandler.websocket().disconnect();
```

Comment:
as I said before, there are formatting issues (I wanted to fix formatting issues when we get an agreement about whole idea), but now it doesn't matter because I removed everything what I developed and added a simple way to initialize layouts, need to get agreement about code design and then we can discuss further steps.